exporter/fileexporter/file_exporter.go

```diff
@@ -112,15 +116,57 @@ func exportMessageAsBuffer(e *fileExporter, buf []byte) error {
 	if err := binary.Write(e.file, binary.BigEndian, data); err != nil {
 		return err
 	}
+
 	return nil
 }
 
+func startFlusher(e *fileExporter) {
+	if e.flushTicker != nil {
+		// flusher is already running.
+		return
+	}
+
+	if e.flushInterval < 1 {
+		// Interval should be 1 or larger.
+		return
+	}
+
+	ff, ok := e.file.(interface{ flush() error })
+	if !ok {
+		// Just in case.
+		return
+	}
+
+	// Start the ticker
+	e.flushTicker = time.NewTicker(time.Second * time.Duration(e.flushInterval))
+	go func() {
+		for e.flushTicker != nil {
+			<-e.flushTicker.C
+			e.mutex.Lock()
+			ff.flush()
+			e.mutex.Unlock()
+		}
+	}()
+}
+
+func stopFlusher(e *fileExporter) {
+	if e.flushTicker == nil {
+		return
+	}
+	e.flushTicker.Stop()
+	e.flushTicker = nil
+}
+
+// Start starts the flush timer.
 func (e *fileExporter) Start(context.Context, component.Host) error {
+	startFlusher(e)
```

Comment:
I would prefer that we determine here whether or not to call `startFlusher` and that we do so by comparing to the user specified value. I think this keeps the business logic clearer.

```go
if e.flushInterval > 0 {
  e.startFlusher()
}
```

Reply:
Changed